♻️ reword the dynamic JSX section (#787)
withastro/docs: src/pages/en/core-concepts/astro-components.md

<p set:html={rawHTMLString} />
```

### JSX Expressions

> You can can define local JavaScript variables inside of the frontmatter component script within an Astro component. You can then inject these variables into the HTML using JSX expressions!

Reviewer: ```suggestion
You can can define local JavaScript variables inside of the frontmatter component script within an Astro component. You can then inject these variables into the component's HTML template using JSX expressions!
```

#### Variables

Local variables can be added into the HTML using the using curly braces syntax:

```astro
---
const name = "Astro";
---
<div>

>   <h1>Hello {name}!</h1> // Outputs <h1>Hello Astro!</h1>

Reviewer: ```suggestion
  <h1>Hello {name}!</h1>  <!-- Outputs <h1>Hello Astro!</h1> -->
```
Updated to be consistent with how we comment out HTML

</div>
```

> #### Attributes

Reviewer: ```suggestion
#### Dynamic Attributes
```
We have some discussions going on specifically re: the use of the words "dynamic" and "interactive"... so I'm not going to approve a change this just yet until some other decisions are made. But, I've made a note of this!

> Local variables can be added into attribute values for both HTML elements and components:

Reviewer: ```suggestion
These local variables can be used in curly braces to pass attribute values to both HTML elements and components:
```

#### Dynamic HTML

> Local variables can be used in JSX-like functions to produce dynamically-generated HTML elements:

Reviewer: ```suggestion
These local variables can be used in JSX-like functions to produce dynamically-generated HTML elements:
```
